### ISSUE TITLE
Listen to the built-in imagesLoaded events

### ISSUE THREAD

**Issue by virgo79** [metafizzy/flickity#328]
Hi David,

I'm using flickity with the built in `imageasLoaded` options set to `true`, and I wish to apply a fadeIn effect each time an image inside a cell is loaded.
Is there a way to listen at the built-in imagesLoaded events?
It could be useful listen to that events, or have a css selector applied to the cell after its content is loaded.
Something similar to the `lazyLoad`  option, where a css selector will be added after a cell is loaded.

Thanks in advance.

**Comment by desandro** [comment]
Sorry, there's no API to hook in to imagesLoaded events. But you can duck-punch one.  Add this code:

``` js
Flickity.prototype.imagesLoaded = function() {
  if ( !this.options.imagesLoaded ) {
    return;
  }
  var _this = this;
  function onImagesLoadedProgress( instance, image ) {
    var cell = _this.getParentCell( image.img );
    _this.cellSizeChange( cell && cell.element );
    if ( !_this.options.freeScroll ) {
      _this.positionSliderAtSelected();
    }
    _this.dispatchEvent( 'imagesLoadedProgress', null, [ image.img, cell.element ] );
  }
  imagesLoaded( this.slider ).on( 'progress', onImagesLoadedProgress );
};
```

This adds a new event `imagesLoadedProgress`.

``` js
// jQuery
$gallery.on( 'imagesLoadedProgress', function( img, cellElement ) {...})
// vanilla JS
flkty.on( 'imagesLoadedProgress', function( img, cellElement ) {...})
```

+ `img` - the loaded Image element
+ `cellElement` - the cell element of the loaded image.

**Comment by virgo79** [comment]
Thank you very much David, this works perfect!

I suppose this could be merged in a future release. Maybe.

I did some tests, and I noticed the loading order of images is random. I mean, that does not follow the order fo the slides, from left to right. 
And probably this depends on many other things, such as connections, http requests, server responses and images weights.

A correct loading sequence could be a very useful update to the flickity plugin, I think.
Actually Im doing this using imagesLoaded as external library and appending each slide to flickity only after its image is full loaded.

Thank you very much again!
